src/modules/compliance/src/lib/Procedure.cpp

```diff
@@ -5,52 +5,147 @@
 
 #include "Logging.h"
 
+#include <cassert>
 #include <parson.h>
-#include <sstream>
 
 namespace compliance
 {
+namespace
+{
+std::size_t SkipSpaces(const std::string& input, std::size_t pos)
+{
+    while (pos < input.size() && isspace(input[pos]))
+    {
+        ++pos;
+    }
+    return pos;
+}
+
+Result<std::size_t> ParseKey(const std::string& input, std::size_t pos)
+{
+    bool first = true;
+    while (pos < input.size() && !isspace(input[pos]) && input[pos] != '=')
+    {
+        if (islower(input[pos]))
+        {
+            return Error("Invalid key: must be uppercase");
```

Comment:
>return Error("Invalid key: must be uppercase"); [](http://example.com/codeflow?start=12&length=47)

Can we fix the case and do not fail, basically become case insensitive? Just a question